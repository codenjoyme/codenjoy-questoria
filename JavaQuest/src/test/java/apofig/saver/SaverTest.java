package apofig.saver;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

/**
 * User: sanja
 * Date: 09.09.13
 * Time: 4:21
 */
public class SaverTest {

    @Test
    public void simple() {
        class D {
            int a, b;
        }

        class C {
            D d;
            int i;
        }

        class B {
            C a;
            C c;
        }

        class A {
            C c;
            B b;
            D d;
        }

        D d = new D();
        d.a = 5;
        d.b = 6;

        C c = new C();
        c.d = d;
        c.i = 4;

        B b = new B();
        b.a = c;
        b.c = new C();

        A a = new A();
        a.b = b;
        a.c = c;
        a.d = d;

        assertEquals(
                "A@0 = {\n" +
                "    b = B@1\n" +
                "    c = C@2\n" +
                "    d = D@3\n" +
                "}\n" +
                "B@1 = {\n" +
                "    a = C@2\n" +
                "    c = C@4\n" +
                "}\n" +
                "C@2 = {\n" +
                "    d = D@3\n" +
                "    i = 4\n" +
                "}\n" +
                "D@3 = {\n" +
                "    a = 5\n" +
                "    b = 6\n" +
                "}\n" +
                "C@4 = {\n" +
                "    d = null\n" +
                "    i = 0\n" +
                "}\n", new Saver().save(a));
    }
}
