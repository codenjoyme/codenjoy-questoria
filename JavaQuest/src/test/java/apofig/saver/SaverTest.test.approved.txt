{"objects":[{"id":"PlayerServiceImpl@0","type":"apofig.javaquest.services.PlayerServiceImpl","fields":[{"game":"JavaQuest@1"},{"players":"HashMap@2"}]},{"id":"JavaQuest@1","type":"apofig.javaquest.map.JavaQuest","fields":[{"initPosition":"PointImpl@4"},{"map":"TerritoryMapImpl@5"},{"objects":"ObjectFactoryImpl@6"},{"players":"LinkedList@7"},{"viewSize":"41"}]},{"id":"HashMap@2","type":"java.util.HashMap","fields":[{"player1-493567632":"Player@3"},{"player2-493567631":"Player@44"}]},{"id":"Player@3","type":"apofig.javaquest.services.Player","fields":[{"game":"JavaQuestSinglePlayer@9"},{"gameCode":"player1-493567632"},{"name":"player1"}]},{"id":"PointImpl@4","type":"apofig.javaquest.map.PointImpl","fields":[{"x":"93"},{"y":"482"}]},{"id":"TerritoryMapImpl@5","type":"apofig.javaquest.map.TerritoryMapImpl","fields":[{"map":"Map@10"},{"objects":"ObjectFactoryImpl@6"},{"fogs":"HashMap@11"},{"height":"564"},{"width":"660"}]},{"id":"ObjectFactoryImpl@6","type":"apofig.javaquest.map.object.ObjectFactoryImpl","fields":[{"loader":"ObjectLoader@13"},{"monsters":"MonsterFactoryImpl@14"},{"objects":"HashMap@15"}]},{"id":"LinkedList@7","type":"java.util.LinkedList","fields":[{"[0]":"Me@8"},{"[1]":"Me@17"}]},{"id":"Me@8","type":"apofig.javaquest.map.object.Me","fields":[{"view":"PlayerView@18"},{"info":"Player@19"},{"whereToGo":null},{"map":"TerritoryMapImpl@5"},{"world":"WorldImpl@16"},{"x":"93"},{"y":"481"}]},{"id":"JavaQuestSinglePlayer@9","type":"apofig.javaquest.map.JavaQuestSinglePlayer","fields":[{"game":"JavaQuest@1"},{"player":"Me@8"}]},{"id":"Map@10","type":"apofig.javaquest.map.Map","fields":[{"map":"char[][]@20"}]},{"id":"HashMap@11","type":"java.util.HashMap","fields":[{"Me@8":"Map@12"},{"Me@17":"Map@42"}]},{"id":"Map@12","type":"apofig.javaquest.map.Map","fields":[{"map":"char[][]@21"}]},{"id":"ObjectLoader@13","type":"apofig.javaquest.map.object.ObjectLoader","fields":[{"except":"ArrayList@22"},{"cache":"HashMap@23"}]},{"id":"MonsterFactoryImpl@14","type":"apofig.javaquest.map.object.monster.MonsterFactoryImpl","fields":[{"monster":null},{"count":"10"},{"monsters":"ArrayList@24"}]},{"id":"HashMap@15","type":"java.util.HashMap","fields":[{"Me@8":"WorldImpl@16"},{"Me@17":"WorldImpl@37"}]},{"id":"WorldImpl@16","type":"apofig.javaquest.map.object.WorldImpl","fields":[{"name":"Me"},{"factory":"ObjectFactoryImpl@6"},{"place":"MapPlace@34"}]},{"id":"Me@17","type":"apofig.javaquest.map.object.Me","fields":[{"view":"PlayerView@35"},{"info":"Player@36"},{"whereToGo":null},{"map":"TerritoryMapImpl@5"},{"world":"WorldImpl@37"},{"x":"93"},{"y":"482"}]},{"id":"PlayerView@18","type":"apofig.javaquest.map.PlayerView","fields":[{"VIEW_CIRCLE":"String[]@38"},{"mask":"char[][]@39"},{"size":"41"},{"vx":"73"},{"vy":"462"}]},{"id":"Player@19","type":"apofig.javaquest.map.Player","fields":[{"name":"player1"},{"gold":"0"}]},{"id":"ArrayList@22","type":"java.util.Arrays$ArrayList","fields":[{"[0]":"class apofig.javaquest.map.object.monster.Monster"},{"[1]":"class apofig.javaquest.map.object.Me"},{"[2]":"class apofig.javaquest.map.object.impl.Nothing"}]},{"id":"HashMap@23","type":"java.util.HashMap","fields":[{"#":"class apofig.javaquest.map.object.impl.Wall"},{"$":"class apofig.javaquest.map.object.impl.Gold"},{"*":"class apofig.javaquest.map.object.impl.dron.Dron"},{"M":"class apofig.javaquest.map.object.impl.dron.DronMentor"}]},{"id":"ArrayList@24","type":"java.util.ArrayList","fields":[{"[0]":"FizzBuzzMonster@25"},{"[1]":"PrimeFactoryMonster@26"},{"[2]":"FibonacciNumbersMonster@27"},{"[3]":"SumSquareDifferenceMonster@28"},{"[4]":"XthPrimeMonster@29"},{"[5]":"PowerDigitSumMonster@30"},{"[6]":"FactorialMonster@31"},{"[7]":"LongDivisionMonster@32"},{"[8]":"MakeBricksMonster@33"}]},{"id":"MapPlace@34","type":"apofig.javaquest.map.MapPlace","fields":[{"map":"Map@10"},{"x":"93"},{"y":"481"}]},{"id":"PlayerView@35","type":"apofig.javaquest.map.PlayerView","fields":[{"VIEW_CIRCLE":"String[]@38"},{"mask":"char[][]@40"},{"size":"41"},{"vx":"74"},{"vy":"462"}]},{"id":"Player@36","type":"apofig.javaquest.map.Player","fields":[{"name":"player2"},{"gold":"0"}]},{"id":"WorldImpl@37","type":"apofig.javaquest.map.object.WorldImpl","fields":[{"name":"Me"},{"factory":"ObjectFactoryImpl@6"},{"place":"MapPlace@41"}]},{"id":"String[]@38","type":"[Ljava.lang.String;","fields":[{"[0]":"????   ????"},{"[1]":"??       ??"},{"[2]":"?         ?"},{"[3]":"?         ?"},{"[4]":"           "},{"[5]":"           "},{"[6]":"           "},{"[7]":"?         ?"},{"[8]":"?         ?"},{"[9]":"??       ??"},{"[10]":"????   ????"}]},{"id":"MapPlace@41","type":"apofig.javaquest.map.MapPlace","fields":[{"map":"Map@10"},{"x":"93"},{"y":"482"}]},{"id":"Map@42","type":"apofig.javaquest.map.Map","fields":[{"map":"char[][]@43"}]},{"id":"Player@44","type":"apofig.javaquest.services.Player","fields":[{"game":"JavaQuestSinglePlayer@45"},{"gameCode":"player2-493567631"},{"name":"player2"}]},{"id":"JavaQuestSinglePlayer@45","type":"apofig.javaquest.map.JavaQuestSinglePlayer","fields":[{"game":"JavaQuest@1"},{"player":"Me@17"}]}]}